packages/permission-controller/src/PermissionController.ts

```diff
@@ -755,6 +764,24 @@ export class PermissionController<
       `${controllerName}:revokePermissions` as const,
       this.revokePermissions.bind(this),
     );
+
+    this.messagingSystem.registerActionHandler(
+      `${controllerName}:updateCaveat` as const,
+      (origin, target, caveatType, caveatValue) => {
+        this.updateCaveat(
```

Comment:
I don't know why, but TS is incorrectly inferring the args to be `string` types, hence the reason I had to pull this into an anon function and cast the types.

Reply:
Could we type the anon function instead of casting? E.g. like `revokeAllPermissions`?

Reply:
TS will complain that it can't assign `string` type to the types I'm casting to for `caveatType` and `caveatValue` if I don't do the casting.

Reply:
Hmm, that's weird. Any idea why @Mrtenz ?

Either way it seems that it is at least only necessary to cast `caveatType`?

Reply:
It's probably collapsing the generic parameters in some way. I don't think generic parameters get carried up by type inference - you need to explicitly declare them.

That won't work here though, because the actions are all declared using `GenericPermissionController`, which is the `PermissionController` type with blank specifications essentially. As a result, we can't really expose the correct types via actions right now. Not without a bunch of refactoring to eliminate `GenericPermissionController` and expose the specification generic params directly on whichever types rely on them.